pkg/downloader/chart_downloader.go

```diff
@@ -143,7 +143,43 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 
 func (c *ChartDownloader) getOciURI(ref, version string, u *url.URL) (*url.URL, error) {
 	var tag string
-	var err error
+
+	registryReference, err := registry.NewReference(u.Path)
+	if err != nil {
+		return nil, err
+	}
+
+	if version == "" {
+		// Use OCI URI tag as default
+		version = registryReference.Tag
+	} else {
+		if registryReference.Tag != "" && registryReference.Tag != version {
+			return nil, errors.Errorf("chart reference and version mismatch: %s is not %s", version, registryReference.Tag)
+		}
+	}
+
+	if registryReference.Digest == "" {
+		if version == "" {
+			return u, errors.Errorf("a digest or version must be specified for an OCI chart")
+		}
```

Comment:
This is a breaking change from today. When no version is passed in Helm will look at the tags and use semver to identify the latest version. This behavior must be preserved to follow the backwards compatibility rules.

Can you update the code to preserve the current behavior when there is no tag, digest, or version present?

Reply:
The error will only be reached if neither digest, `--version` or a tag is not specified.

https://github.com/helm/helm/pull/12690/files#diff-2c9eb8ecd19796ab58f91f47766276c8d2a0c6732276fe1712d47db6d776cfe6R152

Reply:
The only resolution I can think of here is to use `latest` as the tag, but I'm not thrilled about that.

Reply:
The [current functionality](https://github.com/helm/helm/blob/main/pkg/downloader/chart_downloader.go#L153) queries the list of available tags and then selects the most recent semver ordered version. The same logic should be able to be applied in this instance

Reply:
Done.